Tests/ProjLintKitTests/Rules/FileExistenceRuleTests.swift

```diff
@@ -39,4 +39,72 @@ final class FileExistenceRuleTests: XCTestCase {
             XCTAssert(violations.isEmpty)
         }
     }
+
+    func testAllowedPathsWithValidRegex() {
+        resourcesLoaded([infoPlistResource]) {
+            let optionsDict = ["allowed_paths_regex": ["Sources/SuportingFiles/Info\\.plist"]]
+            let rule = FileExistenceRule(optionsDict)
+
+            let violations = rule.violations(in: Resource.baseUrl)
+            XCTAssertEqual(violations.count, 0)
+        }
+
+        resourcesLoaded([infoPlistResource]) {
+            let optionsDict = ["allowed_paths_regex": ["Sources/SuportingFiles/Info2\\.plist"]]
+            let rule = FileExistenceRule(optionsDict)
+
+            let violations = rule.violations(in: Resource.baseUrl)
+            XCTAssertEqual(violations.count, 1)
+            XCTAssertEqual(violations.compactMap { ($0 as? FileViolation)?.url.relativePath }, [infoPlistResource.relativePath])
+            XCTAssertEqual(violations.compactMap { ($0 as? FileViolation)?.message }, ["File exists, but it mustn\'t."])
+        }
+
+        resourcesLoaded([infoPlistResource]) {
+            let optionsDict = ["allowed_paths_regex": ["Sources/SuportingFiles/.*"]]
+            let rule = FileExistenceRule(optionsDict)
+
+            let violations = rule.violations(in: Resource.baseUrl)
+            XCTAssertEqual(violations.count, 0)
+        }
+
+        resourcesLoaded([infoPlistResource]) {
+            let optionsDict = ["allowed_paths_regex": [".*"]]
+            let rule = FileExistenceRule(optionsDict)
+
+            let violations = rule.violations(in: Resource.baseUrl)
+            XCTAssertEqual(violations.count, 0)
+        }
+
+        resourcesLoaded([infoPlistResource]) {
+            let optionsDict = ["allowed_paths_regex": [".*\\.png"]]
+            let rule = FileExistenceRule(optionsDict)
+
+            let violations = rule.violations(in: Resource.baseUrl)
+            XCTAssertEqual(violations.count, 1)
+            XCTAssertEqual(violations.compactMap { ($0 as? FileViolation)?.url.relativePath }, [infoPlistResource.relativePath])
+            XCTAssertEqual(violations.compactMap { ($0 as? FileViolation)?.message }, ["File exists, but it mustn\'t."])
+        }
+    }
+
+    func testAllowedPathsWithInvalidRegex() {
+        let optionsDict = ["allowed_paths_regex": ["["]]
```

Comment:
This string looks weird, maybe in this case we should introduce a variable in the line before and use that instead of the String literal. 😄

Reply:
Done